src/app/question/suggestions/suggestions.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input } from '@angular/core';
+import { QuestionService } from '../question.service';
+
+
+@Component({
+  selector: 'question-suggestions',
+  template: require('./suggestions.component.html'),
+  styles: [
+    require('./suggestions.component.css')
+  ]
+})
+export class QuestionSuggestionsComponent {
+  @Input() currentQuestion: string;
+
+  suggestions: Object[];
+  activated: boolean = true;
+  loading: boolean = false;
+  timeout: boolean = false;
+  missedChange: boolean = false;
+
+  constructor(
+    public questionService: QuestionService,
+  ) {
+  }
+
+  ngOnChanges() {
+    if (!this.timeout) {
+      this.loadSuggestions();
+    } else {
+      this.missedChange = true;
+    }
+  }
+
+  doTimeout() {
+    setTimeout(() => {
```

Comment:
No it's great, but you should listen for the `valueChanges` observable on the input field, debounce it and only then update the model of the `SuggestionComponent`.
Like so:
```js
this.inputFieldInQuestionCmp.valueChanges
      .debounceTime(300)
      .subscribe(val => this.modelThatsBoundToSugesstionCmp = val);
```